src/test/users.test.ts

```diff
@@ -3,55 +3,87 @@ import { AppDataSource } from '../data-source';
 import { queryGetAllUsers } from './query-users';
-import { input } from './constants/input'
+import { input } from './constants/input';
 import { expect } from 'chai';
 import { invalidToken } from './constants/invalid-token';
+import { newUser } from '../utils/data-generate-users';
 
 describe('query users', async () => {
-    beforeEach(async () => {
-        AppDataSource.getRepository(User);
-      });
-      afterEach(async () => {
-        await AppDataSource.getRepository(User).delete({});
-      });
-    
-      it('should return a vector of users', async () => {
-        let arrUsers = []
-        const userOne = {
-            ...input,
-            password: await toHashPassword(input.password)
-          };
-          const user = await addUser(userOne);
-          const token: string = generateToken(user);
-          const response = await queryGetAllUsers(token);  
-          const listUsersResponse = response.data.data.users;
-          delete user.password
-          arrUsers.push(user)
-          expect(listUsersResponse).to.be.deep.equal(arrUsers);
-      })
+  const users = [];
+  let token: string;
 
-      it('should appear an error if token is invalid', async () => {
-        const userOne = {
-          ...input,
-          password: await toHashPassword(input.password)
-        };
-        await addUser(userOne);
-        const response = await queryGetAllUsers(invalidToken);
-        expect(response.data.errors[0].message).to.be.equal('Invalid token');
-        expect(response.data.errors[0].code).to.be.equal(401);
-      });
-    
-      it('an error should appear if authentication is not passed', async () => {
-        const userOne = {
-          ...input,
-          password: await toHashPassword(input.password)
+  beforeEach(async () => {
+    for (let i = 0; i < 20; i++) {
+      users.push(await newUser());
+    }
+
+    AppDataSource.getRepository(User).save(users);
+    const userOne = {
+      ...input,
+      password: await toHashPassword(input.password)
+    };
+
+    const user = await addUser(userOne);
+    users.push(user);
+    token = generateToken(user);
+  });
+
+  afterEach(async () => {
+    AppDataSource.getRepository(User).delete({});
+  });
+
+  it('should return a vector of users', async () => {
+    //ARRANGE
+    const formatArray = users
+      .map((user) => {
+        const userNew = {
+          id: user.id,
+          name: user.name,
+          email: user.email,
+          birthDate: user.birthDate
         };
-        await addUser(userOne);
-        const response = await queryGetAllUsers('');
-        expect(response.data.errors[0].message).to.be.equal(
-          'Authentication required'
-        );
-        expect(response.data.errors[0].code).to.be.equal(401);
+        return userNew;
+      })
+      .sort((a, b) => {
+        const nameA = a.name.toLowerCase();
+        const nameB = b.name.toLowerCase();
+        return nameA < nameB ? -1 : nameA > nameB ? 1 : 0;
       });
```

Comment:
Tem um jeito bem resumido de ordenar alfabeticamente: https://stackoverflow.com/questions/6712034/sort-array-by-firstname-alphabetically-in-javascript